src/library_promise.js

```diff
@@ -260,5 +260,20 @@ mergeInto(LibraryManager.library, {
     dbg(`create: ${id}`);
 #endif
     return id;
-  }
+  },
+
+#if ASYNCIFY == 2
+  emscripten_promise_await_sync__deps: ['$getPromise'],
+  emscripten_promise_await_sync__sig: 'pp',
+  emscripten_promise_await_sync: function(id) {
+    return getPromise(id);
+  },
+#elif ASYNCIFY == 1
+  emscripten_promise_await_sync__deps: ['$getPromise', '$Asyncify'],
+  emscripten_promise_await_sync__sig: 'pp',
+  emscripten_promise_await_sync: function(id) {
+    return Asyncify.handleSleep((wakeUp) => getPromise(id).then(wakeUp));
```

Comment:
`ASYNCIFY == 2` also implements `Asyncify.handleSleep` and we have tended to rely on that compatibility in other places..

Reply:
It's kind of a shame to no take advantage of the fact that JSPI makes this so small and easy, but yeah, having just one version does sound better.

Reply:
Yeah, we should probably look at some kind of general way to take advantage of that..